source/Shader.cpp

```diff
@@ -96,6 +96,10 @@ GLuint Shader::Compile(const char *str, GLenum type)
 	static string version;
 	if(version.empty())
 	{
+#ifdef ES_GLES
+		// Based on GLSL version 3.30
+		version = "#version 300 es";
```

Comment:
good call, it'll take some fancier parsing here, the string I see is `OpenGL ES GLSL ES 3.00 (WebGL GLSL ES 3.00 (OpenGL ES GLSL ES 3.0 Chromium))`

Reply:
Updated, but I haven't found a spec for what this string might look like so I don't know whether this will parse everything thrown at it. Now the code looks farther into the string for a number if necessary and then looks for the substring `"GLSL ES"` to determine if we should add es to the end.

Reply:
The spec suggests it should always begin with a number:
 [OpenGL ES](https://www.khronos.org/registry/OpenGL-Refpages/es3/html/glGetString.xhtml)
[OpenGL](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glGetString.xhtml)

But it appears there is divergence.
In any manner, scanning through the string to find the number is going to be more robust, and it's only done once per session so 🤷‍♂️